sys/log/full/src/log_fcb.c

```diff
@@ -239,45 +217,60 @@ log_fcb_hdr_body_bytes(uint8_t align)
 }
 
 static int
-log_fcb_append_body(struct log *log, const struct log_entry_hdr *hdr,
+log_fcb_append_body(struct log *log, struct log_entry_hdr *hdr,
                     const void *body, int body_len)
 {
-    uint8_t buf[sizeof (struct log_entry_hdr) + LOG_FCB_MAX_ALIGN - 1];
+    uint8_t buf[LOG_BASE_ENTRY_HDR_SIZE + LOG_IMG_HASHLEN + LOG_FCB_MAX_ALIGN - 1];
     struct fcb *fcb;
     struct fcb_entry loc;
     struct fcb_log *fcb_log;
     const uint8_t *u8p;
     int hdr_alignment;
     int chunk_sz;
     int rc;
+    uint16_t ext_len;
 
     fcb_log = (struct fcb_log *)log->l_arg;
     fcb = &fcb_log->fl_fcb;
+    ext_len = 0;
 
     if (fcb->f_align > LOG_FCB_MAX_ALIGN) {
         return SYS_ENOTSUP;
     }
 
-    rc = log_fcb_start_append(log, sizeof *hdr + body_len, &loc);
+    if (hdr->ue_flags & LOG_FLAGS_IMG_HASH) {
+        ext_len = LOG_IMG_HASHLEN;
+    }
```

Comment:
I guess this is another nitpick, but `ext_len` seems like the wrong value to be calculating here.  Every time this variable gets used, it just gets added to `LOG_BASE_ENTRY_HDR_SIZE`.  Rather than manually calculate `ext_len` here, why not just do `hdr_len = log_hdr_len(hdr);`, and use `hdr_len` everywhere in this function.  

`log_fcb_hdr_body_bytes()` could also be modified to take the header length (rather than `ext_len`).  That feels more natural to me.